### PR TITLE
Use NSString + NSRange instead of swift.string + range

### PR DESCRIPTION
**Pull request by schwmi**
For certain asian characters using `attributedString.string[paragraphRange])` leads to an out-of-bounds exception with. 
Screenshot 2019-10-02 at 15 09 35

This PR resolves this by using NSString range operations instead of the String + Range

### CODE REVIEW

**Comment by aquarius** [comment]
@schwmi Do you expect performance regressions with this change?

**Comment by schwmi** [comment]
No, i've tested the encoding performance after the change and it is almost the same (+ 5%)

**Comment by aquarius** [comment]
Thank you. I think it might be a good idea to add a code comment why we use NSString here. Would you remove this change once we drop iOS 12 support? If so we should probably note it somewhere (issue, wiki)

File: Sources/Ashton/AshtonHTMLWriter.swift [hunk 1]
```diff
@@ -31,8 +31,9 @@ final class AshtonHTMLWriter {
                                                     paragraphTag.addAttributes(paragraphStyle)
 
                                                     if nsParagraphRange.length == nsrange.length {
+                                                        let nsString = attributedString.string as NSString
```
Comment (myell0w) [inline Sources/Ashton/AshtonHTMLWriter.swift]: - nit: we have the same variable defined as `string` in the else block - would extract that out of the if/else.
- do we still need `paragraphRange`, if we don't access it here anymore?

**Comment by schwmi** [comment]
@aquarius - i'll add an explanatory comment. Won't remove it as we already had another Range->NSRange bug here (https://github.com/IdeasOnCanvas/Ashton/commit/967903b5f9b24b06e3bb1475ecf68ca2efa3d0b4). Now we use NSRange everywhere. Turns out this is safer if used together with NSAttributedStrings

**Pull request merged**